### PR TITLE
qr code generator

### LINKED ISSUE #30: QR code generation
Implement script which fulfil the following requirements:

1. Code should be placed here https://github.com/ucuapps/robert/blob/kinetic-devel/robert_beverage/scripts/generate_qr_code
2. It should accept the following parameters:
    - x
    - y
    - host
    - port
    - output file name
3. It should print parameters received from command line
4. It should save generated QR code which is URL in form http://host:port/beverage?x=1122&y=222

After starting simulation it could be executed using the following command

```bash
rosrun robert_beverage web_server generate_qr_code
```

### PR DESCRIPTION
**Pull request by v-lavrushko**
closes #30 
Please check my qr code generator program

### CODE REVIEW

**Comment by AndriyPt** [comment]
I am getting this error
```bash
└──╼ rosrun robert_beverage generate_qr_code 
Traceback (most recent call last):
  File "/home/user/workspace/robert/base/src/robert/robert_beverage/scripts/generate_qr_code", line 3, in <module>
    import qrcode
ImportError: No module named qrcode
```
Please read this Tutorial on how dependencies work in ROS http://wiki.ros.org/ROS/Tutorials/rosdep

In your case you could find that ROS by default has this dependency name **python-qrencode** for some qrencode library https://github.com/ros/rosdistro/blob/master/rosdep/python.yaml#L2556
If you specify it in the **package.xml** file of **robert_beverage** package and run these instructions https://github.com/ucuapps/robert/blob/kinetic-devel/CONTRIBUTING.md#code-update new package will be installed.
This is a little bit different library from one which you are using so you need to learn how to call it.
If you have time please read this book about ROS https://www.cse.sc.edu/~jokane/agitr/ :)

**Comment by AndriyPt** [comment]
Also please use **argparse** for arguments parsing https://docs.python.org/2/library/argparse.html

**Review by AndriyPt — CHANGES_REQUESTED** [review]
Please fix my comments

**Comment by v-lavrushko** [comment]
Done

**Review by AndriyPt — CHANGES_REQUESTED** [review]
please remove venv files. It is not needed here. You are using Docker container which is isolated environment itself :)

File: robert_beverage/scripts/generate_qr_code [hunk 1]
```diff
@@ -1,5 +1,15 @@
 #!/usr/bin/env python
 import rospy
```
Comment (AndriyPt) [inline robert_beverage/scripts/generate_qr_code]: please remove this dependency and ros node initialization code. For this utility you do not need ROS node.

File: robert_beverage/scripts/generate_qr_code [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+import qrencode
+import argparse
 
 rospy.init_node('generate_qr_code_node')
-rospy.loginfo('Inside generate_qr_code script')
\ No newline at end of file
+rospy.loginfo('Inside generate_qr_code script')
+
+parser = argparse.ArgumentParser(description='generate qr-code')
+parser.add_argument('data', metavar='D', type=str, nargs=5)
```
Comment (AndriyPt) [inline robert_beverage/scripts/generate_qr_code]: please add separate flag for each option x, y, host, port and filename not single flag DATA :)

**Review by AndriyPt — CHANGES_REQUESTED** [review]

**Comment by v-lavrushko** [comment]
Please check my code

**Comment by LyubomyrD** [comment]
It successfully parsec the following
_python generate_qr_code a b c d e_

I guess, it is worse updating the code to prevent such situation.

**Comment by AndriyPt** [comment]
please run the following commands
```bash
roscd robert_beverage/scripts
rosrun roslint pep8 ./generate_qr_code
```
You will see the following PEP8 errors:
```bash
./generate_qr_code:12:76: E502 the backslash is redundant between brackets
./generate_qr_code:13:50: E127 continuation line over-indented for visual indent
```
Please fix them and check PEP8 rules before requesting review :)

**Comment by v-lavrushko** [comment]
Please, check my code. I checked it for PEP8 this time)

**Comment by LyubomyrD** [comment]
@v-lavrushko The default values of the parameters are not defined. See here https://github.com/ucuapps/robert/issues/30

**Comment by AndriyPt** [comment]
https://docs.python.org/3/library/argparse.html#prog

**Review by AndriyPt — APPROVED** [review]

**Pull request merged**